packages/@lwc/engine-core/src/framework/wiring.ts

```diff
@@ -61,7 +60,16 @@ function createFieldDataCallback(vm: VM, name: string) {
 function createMethodDataCallback(vm: VM, method: (data: any) => any) {
     return (value: any) => {
         // dispatching new value into the wired method
-        invokeComponentCallback(vm, method, [value]);
+        runWithBoundaryProtection(
+            vm,
+            vm.owner,
+            noop,
+            () => {
+                // job
+                method.apply(vm.component, [value]);
```

Comment:
you can use .call instead to avoid creating the extra array.

Reply:
ill need to verify that the method has already the component bind as the `this`.

Reply:
makes sense